packages/jest-cli/src/cli/args.js

```diff
@@ -85,8 +85,9 @@ export const options = {
   bail: {
     alias: 'b',
     default: undefined,
-    description: 'Exit the test suite immediately upon the first failing test.',
-    type: 'boolean',
+    description:
+      'Exit the test suite immediately after `x` number of failing tests.',
+    type: 'number',
```

Comment:
just to verify, you can still do `jest --bail` and it works, right? And what happens if people do `jest --bail my-test-pattern`?

Reply:
It looks like both options still work the same way as they did before.  `jest --bail` will bail after a single failure, and `jest --bail my-test-pattern` also bails after a single failure but only runs the pattern.

Reply:
Awesome!

Reply:
So I was mistaken.  I think I might have been testing with watch mode specifically.  It looks like because `yargs` is expecting a number and there isn't a number specified it just returns undefined for those cases.  So if you just specify `--bail` from the cli it doesn't even recognize that the flag was set and it does nothing.

So I removed the type specifier from `args.js` and I'm now attempting to normalize the data that *might* be in `bail` as a result.